src/go/rpk/pkg/cli/topic/analyze.go

```diff
@@ -0,0 +1,743 @@
+	summary := globalSummary{}
+	parts := 0
+	totalBatches := 0
+	totalDuration := cr.TimeRange.End.Sub(cr.TimeRange.Start)
+	topics := map[string]struct{}{}
+	batchSizes := []int{}
+	batchRates := []float64{}
+
+	for tp, info := range cr.TopicPartitionInfo {
+		parts += 1
+		topics[tp.Topic] = struct{}{}
+
+		if info.empty() {
+			continue
+		}
+
+		batchesPerS := float64(info.BatchesRead) / totalDuration.Seconds()
+		batchRates = append(batchRates, batchesPerS)
+		totalBatches += info.BatchesRead
+
+		if !slices.IsSorted(info.BatchSizes) {
+			slices.Sort(info.BatchSizes)
+		}
+
+		p50, err := utils.Percentile(info.BatchSizes, 50.0)
+		if err != nil {
+			return summary, err
+		}
+		batchSizes = append(batchSizes, p50)
+	}
+
+	summary.Partitions = parts
+	summary.Topics = len(topics)
+	summary.TotalBatchesPerSecond = float64(totalBatches) / totalDuration.Seconds()
+
+	slices.Sort(batchSizes)
+	slices.Sort(batchRates)
+	for _, p := range percentiles {
+		if len(batchRates) != 0 {
+			pBatchRate, err := utils.Percentile(batchRates, p)
+			if err != nil {
+				return summary, err
+			}
+			summary.BatchesPerSecond = append(summary.BatchesPerSecond, percentileValue[float64]{p, pBatchRate})
+		} else {
+			summary.BatchesPerSecond = append(summary.BatchesPerSecond, percentileValue[float64]{p, 0})
+		}
+
+		if len(batchSizes) != 0 {
+			pBatchSize, err := utils.Percentile(batchSizes, p)
+			if err != nil {
+				return summary, err
+			}
+			summary.BatchSizes = append(summary.BatchSizes, percentileValue[int]{p, pBatchSize})
+		} else {
+			summary.BatchSizes = append(summary.BatchSizes, percentileValue[int]{p, 0})
+		}
+	}
+
+	return summary, nil
+}
+
+type printConfig struct {
+	Percentiles        []float64
+	PrintGlobalSummary bool
+	PrintTopicSummary  bool
+}
+
+func (cr *collectResult) printSummaries(cfg printConfig) error {
+	percentiles := cfg.Percentiles
+	percentileNames := []string{}
+	for _, p := range percentiles {
+		percentileNames = append(percentileNames, fmt.Sprintf("P%v", p))
+	}
+
+	const (
+		secSummary   = "summary"
+		secBatchRate = "partition batch rate (1/s)"
+		secBatchSize = "partition batch size (bytes)"
+	)
+
+	ts, err := cr.summarizeTopics(percentiles)
+	if err != nil {
+		return err
+	}
+	// Print the topic summaries in asc order by topic name
+	slices.SortFunc(ts, func(a topicSummary, b topicSummary) int {
+		return strings.Compare(a.Topic, b.Topic)
+	})
+
+	gs, err := cr.summarizeGlobal([]float64{50})
+	if err != nil {
+		return err
+	}
+
+	sections := out.NewMaybeHeaderSections(
+		out.ConditionalSectionHeaders(map[string]bool{
+			secSummary:   cfg.PrintGlobalSummary,
+			secBatchRate: cfg.PrintTopicSummary,
+			secBatchSize: cfg.PrintTopicSummary,
+		})...,
+	)
+
+	sections.Add(secSummary, func() {
+		tw := out.NewTabWriter()
+		tw.PrintColumn("topics", gs.Topics)
+		tw.PrintColumn("partitions", gs.Partitions)
+		tw.PrintColumn("total batch rate (1/s)", gs.TotalBatchesPerSecond)
```

Comment:
I think this is batches/s not 1/s, right? Maybe the row should just be batches/s or batches/second?

Reply:
Yeah, switched it to `batches/s`